AmazingMeSample/app/src/main/java/amazingme/database/FirebaseHelper.java

```diff
@@ -0,0 +1,86 @@
+package amazingme.database;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.util.Log;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+
+import amazingme.activities.app.MainMenu;
+import amazingme.activities.app.RegisterActivity;
+
+public class FirebaseHelper {
+
+    //gonna keep this so we can more easily communicate with the firebase
+
+    private FirebaseDatabase database = FirebaseDatabase.getInstance();
+    private DatabaseReference reference = database.getReference("message");
+
+    public FirebaseHelper(){
+        System.out.println("HERE WE GO");
+        reference.setValue("Hello, world");
+    }
+
+    public static FirebaseAuth getFirebaseAuthInstance() {
+        return FirebaseAuth.getInstance();
+    }
+
+    public static FirebaseUser getFirebaseUser() {
+        return getFirebaseAuthInstance().getCurrentUser();
+    }
+
+    public static void createNewUser(final String firstName,
+                                     final String lastName,
+                                     final String email,
+                                     final String password,
+                                     final Context context) {
+        final FirebaseAuth mAuth = getFirebaseAuthInstance();
+        mAuth.createUserWithEmailAndPassword(email, password)
+            .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                            @Override
+                            public void onComplete(@NonNull Task<AuthResult> task) {
+                                if (task.isSuccessful()) {
+                                    Log.d("SUCCESS", "User Created");
+                                    final Intent loginIntent = new Intent(context, MainMenu.class);
+                                    loginIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
any idea what the flags are for? 

i'd like the info if you've seen it. just wondering if it's necessary

Reply:
So, this issue is part of something bigger that we need to learn/address. We basically cannot simply start a new activity from a non-activity class (notice that this class, FirebaseHelper, is not an activity). This flag is a workaround. We need to eventually standardize startActivity calls.

Reply:
hmmm oh that's good to know. we should talk about that. it might be a workaround for the abstraction the ActivityManager provides until we find something more solid (which lets us use the class in the meantime and just change how it works when we find a better way... god i'm loving this. 

keep this in the meantime.